src/http/request.cr

```diff
@@ -88,6 +107,14 @@ class HTTP::Request
     HTTP::Params.parse(uri.query || "")
   end
 
+  private def parse_body_params
+    unless has_body_params?
+      raise "Content-Type should be #{BODY_PARAMS_CONTENT_TYPE} to use #body_params"
```

Comment:
That was my first take - create custom exception. Basically I didn't find any custom exceptions in `http` std lib, so I was reluctant to put any.

`ArgumentError` sounds good to me too.

3rd option - can we avoid exception at all in this case? - that is what I hope for.